src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -303,7 +303,107 @@ class EngineSegment(BaseSegment):
         "ENGINE",
         Ref("EqualsSegment"),
         Sequence(
-            Ref("EngineFunctionSegment"),
+            Ref("TableEngineFunctionSegment"),
+            AnySetOf(
+                Sequence(
+                    "ORDER",
+                    "BY",
+                    OneOf(
+                        Ref("BracketedColumnReferenceListGrammar"),
+                        Ref("ColumnReferenceSegment"),
+                    ),
+                    optional=True,
+                ),
+                Sequence(
+                    "PARTITION",
+                    "BY",
+                    Ref("ExpressionSegment"),
+                    optional=True,
+                ),
+                Sequence(
+                    "PRIMARY",
+                    "KEY",
+                    Ref("ExpressionSegment"),
+                    optional=True,
+                ),
+                Sequence(
+                    "SAMPLE",
+                    "BY",
+                    Ref("ExpressionSegment"),
+                    optional=True,
+                ),
+                Sequence(
+                    "SETTINGS",
+                    Delimited(
+                        AnyNumberOf(
+                            Sequence(
+                                Ref("NakedIdentifierSegment"),
+                                Ref("EqualsSegment"),
+                                OneOf(
+                                    Ref("NumericLiteralSegment"),
+                                    Ref("QuotedLiteralSegment"),
+                                ),
+                                optional=True,
+                            ),
+                        )
+                    ),
+                    optional=True,
+                ),
+            ),
+        ),
+    )
+
+
+class DatabaseEngineFunctionSegment(BaseSegment):
+    """A ClickHouse `ENGINE` clause function.
+
+    With this segment we attempt to match all possible engines.
+    """
+
+    type = "engine_function"
+    match_grammar: Matchable = Sequence(
+        Sequence(
+            # Ref(
+            #     "FunctionNameSegment",
+            #     exclude=OneOf(
+            #         Ref("DatePartFunctionNameSegment"),
+            #         Ref("ValuesClauseSegment"),
+            #     ),
+            # ),
+            OneOf(
+                "Atomic",
```

Comment:
Can we please follow the convention of uppercase keywords here too?

Reply:
Fixed in commit `f56765e`.